xformers/components/positional_embedding/rotary.py

```diff
@@ -73,7 +78,10 @@ def forward(
         self, q: torch.Tensor, k: torch.Tensor
     ) -> Tuple[torch.Tensor, torch.Tensor]:
         self._cos_cached, self._sin_cached = self._update_cos_sin_tables(
-            q, seq_dimension=-2
+            k, seq_dimension=-2
         )
 
-        return apply_rotary_pos_emb(q, k, self._cos_cached, self._sin_cached)
+        return (
```

Comment:
side effect, out of principle rotary supports different k/q dimensions but the previous implementation would choke on it, fixing that